### PR TITLE
Add `@Buildable` annotation to all model classes for generating the Builder classes

### PR DESCRIPTION
**Pull request by im-konge**
This PR adds `@Buildable` annotation to the model Classes, so the `Builder`, `Fluent`, and `FluentImpl` classes are generated during the project build.
The `Builder` classes are used for testing purposes, but also for creating the `KafkaAccess` CR using Java code (and not just YAMLs).
Finally, it changes the tests to use these new builders.

### CODE REVIEW

**Review by see-quick — COMMENTED** [review]
I assume the second PR would be the modification of tests?

**Comment by im-konge** [comment]
> I assume the second PR would be the modification of tests?

Yes, I didn't want to include this in #48 and have it separated to other stuff.
So I wanted to do the change in all classes in different PR.

File: pom.xml
Comment (scholzj) [inline pom.xml:307]: This sucks because it disables Javadocs checks for all other classes as well.

**Review by scholzj — COMMENTED** [review]
Doesn't Fabric8 have its own builders for the CRDs?

File: pom.xml
Comment (im-konge) [reply pom.xml:307]: I know
But in case of the Builder, Fluent and FluentImpl classes, there are Javadoc warnings that will fail the build.
I was thinking about having the separate module just for these resources (the `api` module) and then the operator module, to split the things and manipulate it better in the children pom.xmls

**Comment by im-konge** [comment]
> Doesn't Fabric8 have its own builders for the CRDs?

The Fabric8 has deps for building the CRDs (from POJOs) and the POJOs (from CRDs), but based on articles I read, these can be extended by Sundrio's extensions and annotations (because the deps are using Sundrio underneath) to generate the Builders.

I will check that more, but from what I saw until now, that is the way how it is used now (but maybe I'm completely wrong of course).

File: pom.xml
Comment (im-konge) [reply pom.xml:307]: @scholzj @katheris @mstruk WDYT about having two modules and splitting the models with the operator's code?

Comment (mstruk) [reply pom.xml:307]: IIUC separating out the `api` module would be like the Strimzi operator project does it. Also, I guess we need it this way in order to have two different warning treatments. Sounds to me like the way to go.

Comment (katheris) [reply pom.xml:307]: I prefer the suggestion of having a separate `api` module similar to the cluster-operator. I guess it also distinguishes which classes we might want to encourage users to depend on if they were programmatically creating `KafkaAccess` resources

**Comment by katheris** [comment]
Other than the discussion about preventing javadoc warnings from failing the build these changes look good to me. I'll hold off approving until that is resolved though

File: pom.xml
Comment (im-konge) [reply pom.xml:307]: Cool, so I will add the `api` module there :) thanks

**Comment by im-konge** [comment]
@scholzj @katheris @mstruk I added two modules - `api` and `operator` - and changed everything related to that accordingly. Could you please have another look? Thanks

**Review by scholzj — APPROVED** [review]
LGTM assuming it wortks.

File: .azure/templates/jobs/build_java.yaml
Comment (katheris) [inline .azure/templates/jobs/build_java.yaml:27]: As far as I can see this isn't being invoked yet, is the expectation that it will be needed when we add the system tests?

**Comment by katheris** [comment]
@im-konge what was the reasoning for putting StatusUtils and StatusUtilsTest inside the `api` module? I think because the status is only updated by the operator, and the methods there are more around updating the status rather than inspecting it, I would expect those to stay with the other internal classes in the `operator` module.

I'm also wondering whether the `internal` package should be renamed to something else, I guess it contains classes for parsing the Kafka CR, but I can't come up with a better name, so happy to leave it as internal for now and we can always rename in future.

File: .azure/templates/jobs/build_java.yaml
Comment (scholzj) [reply .azure/templates/jobs/build_java.yaml:27]: I think that is because the PR is from fork.

Comment (im-konge) [reply .azure/templates/jobs/build_java.yaml:27]: It is needed because of the spotbugs plugin, which needs the modules to be build before it runs the checks -> as you will get the `dependency issue found` type of error otherwise.

**Comment by im-konge** [comment]
@katheris the reason I moved the `StatusUtils` there is because those methods are used inside the `KafkaAccessStatus`, which should be part of the `api` module. In case that I would use it in `operator` module, I would create a cyclic dependency.

Maybe it should not be there and it should be done a bit differently, but I guess that's something that should be changed in some other PR (I'm fine with changing it, but I don't want to make the scope of the PR bigger and bigger).

> I'm also wondering whether the internal package should be renamed to something else, I guess it contains classes for parsing the Kafka CR, but I can't come up with a better name, so happy to leave it as internal for now and we can always rename in future.

Sure I can change it as well if there is a better name.

File: .checkstyle/suppressions.xml
Comment (im-konge) [reply .checkstyle/suppressions.xml:14]: thanks, it seems like I accidentally smashed the keyboard
good catch anyway :D

File: .azure/templates/jobs/build_java.yaml
Comment (im-konge) [reply .azure/templates/jobs/build_java.yaml:27]: The exact error is shown here: https://dev.azure.com/cncf/strimzi/_build/results?buildId=171513&view=logs&j=2de397de-21fc-5e8b-587d-4865bafb3e06&t=d9c95e40-2ff1-577f-4ff1-10fe2890ceac

```
[ERROR] Failed to execute goal on project operator: Could not resolve dependencies for project io.strimzi.kafka.access:operator:jar:0.0.1-SNAPSHOT: Could not find artifact io.strimzi.kafka.access:api:jar:0.0.1-SNAPSHOT -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal on project operator: Could not resolve dependencies for project io.strimzi.kafka.access:operator:jar:0.0.1-SNAPSHOT: Could not find artifact io.strimzi.kafka.access:api:jar:0.0.1-SNAPSHOT

```

**Comment by katheris** [comment]
@im-konge ah thanks, that makes sense, yeah lets leave it there for now

**Review by katheris — APPROVED** [review]

File: .azure/templates/jobs/build_java.yaml
Comment (katheris) [reply .azure/templates/jobs/build_java.yaml:27]: Ok, thanks

**Comment by katheris** [comment]
Thanks @im-konge for the PR, it looks good to me. I'll leave it a day or so to see if @mstruk has any comments, and if not will go ahead and merge it.

**Pull request merged**
